pkg/cac/cac.go

```diff
@@ -2,25 +2,59 @@ package cac
 
 import (
 	"encoding/binary"
+	"errors"
 
 	"github.com/ethersphere/bee/pkg/bmtpool"
 	"github.com/ethersphere/bee/pkg/swarm"
 )
 
+var (
+	errTooShortChunkData = errors.New("short chunk data")
+	errTooLargeChunkData = errors.New("data too large")
+)
+
+// New creates a new content address chunk by initializing a span and appending the data to it.
 func New(data []byte) (swarm.Chunk, error) {
-	hasher := bmtpool.Get()
-	defer bmtpool.Put(hasher)
+	span := make([]byte, swarm.SpanSize)
```

Comment:
you should do the input validation the same way you do in `NewWithDataSpan`, since `data` can be larger than a single chunk, which should result in an error